components/InvoiceFinancing/index.tsx

```diff
@@ -163,59 +164,71 @@ function ApiInvoices() {
   }
 
   // hard coded for now
-  const onCreateInvoiceClick = () => {
-    const newInvoiceNumber = getNewInvoiceNumber()
+  const onCreateInvoiceClick = async () => {
+    setIsLoading(true)
```

Comment:
this is almost instantaneous. there is no I/O. so i don't think we need to set loading here